### DISCUSSION TITLE
0.5.1 Release Timeline

### DISCUSSION THREAD

**Discussion by nilsleh** [torchgeo/torchgeo#1709] [Announcements]
We're planning a 0.5.1 release next week. This will include all bug fixes in [0.5.1](https://github.com/microsoft/torchgeo/milestone/15).

If you have any bugs you would like to see fixed, now is the time to submit PRs!

**Comment by nilsleh** [comment]
This is a bugfix release. There are no new features or API changes with respect to the 0.5.0 release.

## Datamodules

- EuroSAT: make channel normalization statistics responsive to dynamic band selection (#1634, #1681)

## Datasets

- AGB Live Woody Biomass: update download link for dataset (#1679, #1713)
- EuroSAT: remove classes attribute and instead rely on `ImageFolder` classes (#1648, #1650)
- OSCD: change image datatype be float instead of int (#1652, #1656)
- RESICS45: remove classes attribute and instead rely on `ImageFolder` classes (#1648, #1650)
- UC Merced: fix plotting which expects images from dataset to be normalized already (#1712)
- UC Merced: remove classes attribute and instead rely on `ImageFolder` classes (#1648, #1650)
- GeoDataset: check if the path points to a Virtual File System, to prevent error of looking and not finding the paths locally (#1605, #1612)
- GeoDatasets: consistent use of `paths` argument instead of `root` in `RuntimeError` of several datasets (#1704, #1717)

## Trainers

- During logging, trainers were expecting a datamodule with plot functionality, which was preventing trainers from being used with custom Pytorch Dataloaders (#1703)
- Remove default callback configurations of trainers and leave it to user instead (#1640, #1641, #1642, #1645, #1647)
- Skip weights and augmentations when saving hparams, allowing these parameters to be changed (#1622, #1639, #1670)

## Scripts

- Solve logging conflict by allowing `config.yaml` file to be overwritten (#1621, #1625)

## Tests

- Greatly reduce memory footprint of CI which was causing PR tests to fail (#1658)
- Copy testing csv file instead of downloading it for MapInWild dataset test (#1657)
- Fix `choco install unrar` in CI by using `7zip` instead of `unrar` (#1697)
- CI: use unique names for release caches (#1723)

## Documentation

- README: update `SemanticSegmentationTask` example with arguments introduced in 0.5 (#1608)
- README: add section on LightningCLI usage with torchgeo (#1626, #1628)
- README: add section on availability of pretrained weights in torchgeo (#1716)
- BioMassters: fix typo in docs' overview table of non-geo datasets (#1718)
- SSL4EO-L Benchmark: add dataset information to documentation (#1719)

## Contributors

This release is thanks to the following contributors (in alphabetical order):

@adamjstewart
@ashnair1
@dylanrstewart
@kaybe20
@menglutao
@nilsleh
@pioneerHitesh
@robmarkcole
